helpers/chatHelper.py

```diff
@@ -266,462 +264,629 @@ async def partChannel(
+            "token_id": user_token["token_id"],
+            "username": user_token["username"],
+            "user_id": user_token["user_id"],
+            "send_to": send_to,
+        },
+    )
+
+
+async def sendMessage(
+    token_id: str,
+    send_to: str,
+    message: str,
+) -> Optional[int]:
+    user_token = await osuToken.get_token(token_id)
+    if user_token is None:
+        logger.warning(
+            "User tried to send message but they are not connected to server",
+            extra={"token_id": token_id},
+        )
+        return ...  # TODO: return IRC codes
+
+    if user_token["user_id"] == CHATBOT_USER_ID:
+        return await handle_bot_message(
+            user_token=user_token,
+            send_to=send_to,
+            message=message,
+        )
+
+    if user_token["tournament"]:
+        logger.warning(
+            "User tried to send message but they are connected from tournament client",
+            extra={
+                "token_id": token_id,
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "send_to": send_to,
             },
         )
-        return 404
-    except exceptions.invalidArgumentsException:
+        return ...
+
+    if osuToken.is_restricted(user_token["privileges"]):
         logger.warning(
-            "User tried to send an invalid message",
+            "User tried to send message but they are in restricted mode",
             extra={
-                "username": userToken["username"],
-                "user_id": userToken["user_id"],
-                "recipient": to,
+                "token_id": token_id,
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "send_to": send_to,
             },
         )
-        return 404
-    except:
-        logger.exception("An unhandled exception occurred whle sending a chat message")
+        return ...
+
+    if await osuToken.isSilenced(user_token["token_id"]):
+        silence_time_left = await osuToken.getSilenceSecondsLeft(user_token["token_id"])
+        await osuToken.enqueue(
+            user_token["token_id"],
+            serverPackets.silenceEndTime(silence_time_left),
+        )
+
+        logger.warning(
+            "User tried to send message during silence",
+            extra={
+                "token_id": token_id,
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "send_to": send_to,
+            },
+        )
+        return ...
+
+    if not message.strip():
+        logger.warning(
+            "User tried to send an empty message",
+            extra={
+                "token_id": token_id,
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "send_to": send_to,
+            },
+        )
+        return ...
+
+    # optimise message
```

Comment:
"optimise" doesn't really make sense in the context, the intention here is a security check
```suggestion
    # truncate long messages
```